### PR TITLE
fix(grpc): refactor kserve inference dispatch and fix masked model-not-found errors

### LINKED ISSUE #10274: [CONTRIBUTION]: fix(grpc): refactor kserve inference dispatch and fix masked model-not-found errors
### Type of Change

Bug fix

### Problem Statement

Two issues exist in `lib/llm/src/grpc/service/kserve.rs`:

1. When a completions model is not registered, both `model_infer` and `model_stream_infer` fall through to `request.try_into()` which produces a misleading "Failed to parse request" error instead of a clear not-found response. This masks the true cause from operators debugging a failed inference request.

2. The tensor/completion dispatch logic and request template application are duplicated independently in both `model_infer` and `model_stream_infer`, noted by two `[gluo TODO] refactor to reuse code` comments.

### Proposed Solution

- Add `is_completions_model` to `State` and use it as an early guard in request conversion, returning `Status::not_found` with the correct model name before `try_into()` is ever called
- Extract `InferRequest` enum with `from_model_infer`, `unary_response`, and `stream_response` to centralise dispatch logic shared between both handlers
- Extract `apply_request_template` helper to remove the duplicated template application block
- Use `resolve_request_model` before the existence check so error messages reflect the template-resolved model name
- Add unit tests covering template application, the not-found error, and the template-resolved name edge case

### Estimated PR Size

L (201-500 lines)

### Files/Components Affected

`lib/llm/src/grpc/service/kserve.rs`

### PR DESCRIPTION
**Pull request by jh-nv**
## Summary
Fixes two issues in `lib/llm/src/grpc/service/kserve.rs`:

1. **Masked model-not-found errors.** When a completions model is not registered, both `model_infer` and `model_stream_infer` fell through to `request.try_into()`, surfacing a misleading `InvalidArgument` "Failed to parse request" instead of a clear not-found response. Now `from_model_infer` checks existence via a new `State::is_completions_model` guard — against the **template-resolved** model name (`resolve_request_model`) — and returns `Status::not_found` naming the model before any parse step.
2. **Duplicated dispatch logic.** The tensor/completion dispatch and request-template application were duplicated in both handlers (two `[gluo TODO] refactor to reuse code` comments). Centralized into an `InferRequest` enum (`from_model_infer` / `unary_response` / `stream_response`) and an `apply_request_template` helper.

## Test plan
- `cargo build/test/clippy/fmt -p dynamo-llm` — green (clippy `-D warnings`).
- New unit tests: template application, the not-found error, and the template-resolved-name edge case.
- **Container-validated over real gRPC:** launched the KServe gRPC frontend and sent a `ModelInfer` for an unregistered model → `NOT_FOUND "Model 'does-not-exist-model' not found"` (previously the masked parse error).

Resolves ai-dynamo/dynamo#10274 (Linear DGH-938).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://nvidia.devinenterprise.com/review/ai-dynamo/dynamo/pull/10621" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling for missing model requests with clearer `NotFound` status messages.
  * Early model validation prevents missing-model errors from being masked by request parsing failures.

* **Tests**
  * Added comprehensive validation tests for model request handling and error scenarios.

### CODE REVIEW

File: lib/llm/src/grpc/service/kserve.rs [hunk 1]
```diff
@@ -315,86 +319,270 @@ impl Config {
...
+    if let Some(template) = template {
+        if completion_request.inner.model.is_empty() {
+            completion_request.inner.model = template.model.clone();
+        }
+        if completion_request.inner.temperature.unwrap_or(0.0) == 0.0 {
+            completion_request.inner.temperature = Some(template.temperature);
+        }
+        if completion_request.inner.max_tokens.unwrap_or(0) == 0 {
+            completion_request.inner.max_tokens = Some(template.max_completion_tokens);
+        }
```
[the remark this answers is not in the document]
Comment (jh-nv) [reply lib/llm/src/grpc/service/kserve.rs]: Fixed in 262fb63f80 — switched to `is_none()` so an explicit `temperature = 0.0` (deterministic) or `max_tokens = 0` is preserved rather than treated as unset, matching the function's documented contract. Added a test asserting explicit zeros survive the template merge.

File: lib/llm/src/grpc/service/kserve.rs [hunk 2]
```diff
@@ -315,86 +319,270 @@ impl Config {
...
+) {
+    if let Some(template) = template {
+        if completion_request.inner.model.is_empty() {
+            completion_request.inner.model = template.model.clone();
+        }
+        if completion_request.inner.temperature.unwrap_or(0.0) == 0.0 {
+            completion_request.inner.temperature = Some(template.temperature);
+        }
+        if completion_request.inner.max_tokens.unwrap_or(0) == 0 {
+            completion_request.inner.max_tokens = Some(template.max_completion_tokens);
```
[the remark this answers is not in the document]
Comment (jh-nv) [reply lib/llm/src/grpc/service/kserve.rs]: Fixed in 262fb63f80 — `apply_request_template` now uses `is_none()` for temperature and max_tokens, so explicit `Some(0.0)`/`Some(0)` are no longer clobbered by the template. Added test coverage for the explicit-zero case.

**Review by GuanLuo — APPROVED** [review]
